api/models/Post.js

```diff
@@ -236,18 +236,24 @@ module.exports = bookshelf.Model.extend({
         var users = uniqById(_.flatten(usersWithDupes))
 
         _.remove(users, user => user.get('id') === creator.get('id'))
-        return Promise.map(users, (user) => {
-          if (!user.get('push_new_post_preference')) return
-          if (post.isWelcome()) return
-          var userCommunities = user.relations.communities.models
-          var postCommunitiesIds = communities.models.map(community => community.get('id'))
-          var community, path, alertText
-          community = _.find(userCommunities, community => _.contains(postCommunitiesIds, community.get('id')))
-          if (!community) return
-          path = url.parse(Frontend.Route.post(post, community)).path
-          alertText = PushNotification.textForNewPost(post, community, user.get('id'))
-          return user.sendPushNotification(alertText, path)
-        })
+        return Promise.join(
+          Promise.map(communities.models, (community) => {
```

Comment:
just a little style note: you don't need to have `( )` for single-argument methods